packages/web/src/components/edit/fields/visibility/VisibilityField.test.tsx

```diff
@@ -0,0 +1,150 @@
+      }
+    ]
+  }
+
+  return render(
+    <Formik
+      initialValues={merge(defaultInitialValues, initialValues)}
+      onSubmit={submitSpy}
+    >
+      <Form>
+        <VisibilityField entityType='track' />
+        <button type='submit'>Submit</button>
+      </Form>
+    </Formik>
+  )
+}
+
+describe('VisibilityField', () => {
+  beforeEach(() => {
+    vi.useFakeTimers()
+    submitSpy.mockClear()
+  })
+
+  afterEach(() => {
+    vi.useRealTimers()
+  })
+
+  it('renders visibility field with "public" initial value', () => {
+    renderTrackVisibilityField()
+
+    expect(
+      screen.getByRole('heading', { name: 'Visibility' })
+    ).toBeInTheDocument()
+
+    expect(
+      screen.getByText(
+        'Change the visibility of this release or schedule it to release in the future.'
+      )
+    ).toBeInTheDocument()
+
+    expect(screen.getByLabelText(/public/i)).toBeInTheDocument()
+  })
+
+  it('clicking opens a dialog with visibility options', () => {
+    renderTrackVisibilityField()
+
+    expect(
+      screen.getByRole('button', { name: /visibility/i })
+    ).toBeInTheDocument()
+
+    fireEvent.click(screen.getByRole('button', { name: /visibility/i }))
+
+    expect(
+      screen.getByRole('dialog', { name: /visibility/i })
+    ).toBeInTheDocument()
+
+    expect(screen.getByRole('radio', { name: /public/i })).toBeInTheDocument()
+    expect(screen.getByRole('radio', { name: /hidden/i })).toBeInTheDocument()
+    expect(
+      screen.getByRole('radio', { name: /scheduled release/i })
+    ).toBeInTheDocument()
+  })
+
+  it('can set track entity as unlisted', async () => {
+    renderTrackVisibilityField()
+
+    expect(
+      screen.getByRole('button', { name: /visibility/i })
+    ).toBeInTheDocument()
+
+    fireEvent.click(screen.getByRole('button', { name: /visibility/i }))
+
+    fireEvent.click(screen.getByRole('radio', { name: /hidden/i }))
+
+    fireEvent.click(screen.getByRole('button', { name: /save/i }))
+
+    await waitForElementToBeRemoved(() => screen.queryByRole('dialog'))
+
+    expect(await screen.findByLabelText(/hidden/i)).toBeInTheDocument()
+
+    fireEvent.click(screen.getByRole('button', { name: /submit/i }))
+
+    await waitFor(() => {
+      expect(submitSpy).toHaveBeenCalledWith(
+        expect.objectContaining({
+          trackMetadatas: [
+            {
+              is_scheduled_release: false,
+              is_unlisted: true,
+              release_date: ''
+            }
+          ]
+        }),
+        expect.anything()
+      )
+    })
+  })
+
+  it.only('can set as a scheduled release', async () => {
+    renderTrackVisibilityField()
+
+    fireEvent.click(screen.getByRole('button', { name: /visibility/i }))
+    fireEvent.click(screen.getByRole('radio', { name: /scheduled release/i }))
+
+    fireEvent.click(screen.getByRole('button', { name: /release date/i }))
+    fireEvent.click(
+      screen.getByRole('button', { name: 'Saturday, June 15, 2024' })
```

Comment:
is this test still going to work in July?

Reply:
:D yeah it will cause a mocked the dates earlier in this file :)

Reply:
ah, yes. Sweet!